### DISCUSSION TITLE
New found about “This slug conflicts with an existing URL.”

### DISCUSSION THREAD

**Discussion by liuzy2008** [django-wiki/django-wiki#1288] [General]
I found ```urlpatterns += static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)``` would cause “This slug conflicts with an existing URL.” problem. Hope this would help

**Comment by ghost-ng** [comment]
what is the work around then?  doesnt the app need those paths to serve the static files?  what if I have another apps alongside it that require those paths?
@benjaoming

**Reply by benjaoming** [reply]
It sounds like a misconfiguration - firstly, this configuration is only for development environments. Secondly, the `MEDIA_URL` shouldn't overlap with the path in which wiki articles reside.

Would need more information to understand this. If you experience a bug, please describe the classic pattern of what you're expecting to happen vs. what happens.

**Reply by ghost-ng** [reply]
thanks for the reply - I am apparently stumbling on the age old "This slug conflicts with an existing URL" issue.

I have a vanilla install along with an already active project with other side-by-side app.  The install is a <>/wiki/ which routes perfectly.  I can actually create the first root article just fine too.

install:
`pip install git+https://github.com/django-wiki/django-wiki.git
`

I then followed these instructions nearly to the T - https://django-wiki.readthedocs.io/en/main/installation.html (settings, db migrate, URLpatterns)

When trying to create child page:
<img width="658" alt="image" src="https://github.com/django-wiki/django-wiki/assets/21272239/f49326f5-13b0-40f1-bbf7-6e93c759797b">


I then following some common solutions like checking overly broad URL patterns and setting the slug check to false, neither work
I of course can check my URLs again but this is 100% my only app and URL routing for <root>/wiki/
From settings.py:
```
SITE_ID = 1
CHECK_SLUG_URL_AVAILABLE=False
```

urls.py:
```
urlpatterns = [
    path('admin/', admin.site.urls),
    path('notifications/', include('django_nyt.urls')),
    path('wiki/', include('wiki.urls')),
    path('login/', auth_views.LoginView.as_view(template_name='login.html'), name='login'),
    path('logout/', CustomLogoutView.as_view(), name='logout'),
   ...
   ...
] + static(settings.STATIC_URL, document_root=settings.STATIC_ROOT) + static(settings.MEDIA_URL, document_root=settings.MEDIA_ROOT)
```

I didn't dig too much into the code, but it does seem odd to me that the root article didn't get a proper slug, but maybe that is expected/irrelevent
<img width="502" alt="image" src="https://github.com/django-wiki/django-wiki/assets/21272239/f57ce04f-aab6-43fb-9f70-003c81a7cd78">

**Reply by ghost-ng** [reply]
random thought that just occurred to me, if my app is at /wiki/ shouldn't my slug start with /wiki/ instead of the root (simply just '/') - it doesn't seem like I can alter that on the page.
@benjaoming

**Reply by benjaoming** [reply]
In order to disable the check, your setting should be named `WIKI_CHECK_SLUG_URL_AVAILABLE`, not `CHECK_SLUG_URL_AVAILABLE`

LMK how it goes :)

**Reply by ghost-ng** [reply]
omg - yea that works now - so what does this check actually do then because I was afraid I would get URL pattern collisions, but this fix actually seems safe.  I tried to create a collision but the slug check is obviously working still (but yes now I can create subpages - either way, I feel dumb, and thank you!)
<img width="658" alt="image" src="https://github.com/django-wiki/django-wiki/assets/21272239/a4a58f32-bcba-48a4-886e-0d8f88c81cb3">

**Reply by benjaoming** [reply]
You can try to access the address that is reported as occupied and see which django view responds? i.e. go to `http://localhost:8000/wiki/skljfksldf` and see what responds.

**Reply by ghost-ng** [reply]
the solution you had worked perfectly, im more asking out of curiosity now - the slug checks are clearly still happening, but now I'm at least able to create posts.  Under the hood, why is this the case?

**Reply by benjaoming** [reply]
To not conflict with an existing page

But please try the experiment, it's one way of telling if it was a false negative or that you indeed have a view that returns something.
